dev/devicelab/lib/tasks/analysis.dart

```diff
@@ -28,76 +28,101 @@ Future<TaskResult> analyzerBenchmarkTask() async {
     await dart(<String>['dev/tools/mega_gallery.dart', '--out=${_megaGalleryDirectory.path}']);
   });
 
-  final Map<String, dynamic> data = <String, dynamic>{
-    'flutter_repo_batch': await _run(new _FlutterRepoBenchmark()),
-    'flutter_repo_watch': await _run(new _FlutterRepoBenchmark(watch: true)),
-    'mega_gallery_batch': await _run(new _MegaGalleryBenchmark()),
-    'mega_gallery_watch': await _run(new _MegaGalleryBenchmark(watch: true)),
-  };
+  final Map<String, dynamic> data = <String, dynamic>{};
+  data.addAll((await _run(new _FlutterRepoBenchmark())).asMap('flutter_repo', 'batch'));
+  data.addAll((await _run(new _FlutterRepoBenchmark(watch: true))).asMap('flutter_repo', 'watch'));
+  data.addAll((await _run(new _MegaGalleryBenchmark())).asMap('mega_gallery', 'batch'));
+  data.addAll((await _run(new _MegaGalleryBenchmark(watch: true))).asMap('mega_gallery', 'watch'));
 
   return new TaskResult.success(data, benchmarkScoreKeys: data.keys.toList());
 }
 
-/// Times how long it takes to analyze the Flutter repository.
-class _FlutterRepoBenchmark {
-  _FlutterRepoBenchmark({ this.watch = false });
+class _BenchmarkResult {
+  const _BenchmarkResult(this.mean, this.min, this.max);
 
-  final bool watch;
+  final double mean; // seconds
 
-  Future<double> call() async {
-    section('Analyze Flutter repo ${watch ? 'with watcher' : ''}');
-    final Stopwatch stopwatch = new Stopwatch();
-    await inDirectory(flutterDirectory, () async {
-      final List<String> options = <String>[
-        '--flutter-repo',
-        '--benchmark',
-      ];
+  final double min; // seconds
 
-      if (watch)
-        options.add('--watch');
+  final double max; // seconds
 
-      stopwatch.start();
-      await flutter('analyze', options: options);
-      stopwatch.stop();
-    });
-    return stopwatch.elapsedMilliseconds / 1000;
+  Map<String, dynamic> asMap(String benchmark, String mode) {
+    return <String, dynamic>{
+      '${benchmark}_$mode': mean,
```

Comment:
If we send the mean back as the main value to graph, this will introduce more noise into the benchmark - I believe `min` will be the most stable value. I really do want to reduce noise here, so that we can better identify regressions, and know when any performance work we do has a measurable effect.

Perhaps we can duplicate the graphs, and track median as well as min.

Reply:
This patch will create graphs for all three (twelve, really).